docs/user-guide/inState.md

```diff
@@ -0,0 +1,30 @@
+# inState`<State>`
+
+The first concept of the DSL we learn is `inState`:
+
+```kotlin
+class ItemListStateMachine(
+    private val httpClient: HttpClient
+) : FlowReduxStateMachine<ListState, Action>(initialState = Loading) {
+
+    init {
+        spec {
+            inState<Loading> {
+                ...
+            }
+        }
+    }
+}
+```
+
+`inState<Loading>` is just an "entry point".
```

Comment:
I think it would make sense to add a sentence here that actually says what `inState` actually means. Something like
```suggestion
`inState<Loading>` is just an "entry point" that describes that anything inside this block should only be executed while the state machine is currently in a state that matches the given class.
```